### PR TITLE
switch to follow_redirects on httpx.get call in CloudSQL provider

### PR DESCRIPTION
**Pull request by philipherrmann**
Also:
- set min httpx version to 0.21.1 to match follow_redirects API
- fix weakness in _get_bool which prevented to explicitly set parameters to 'false' in extras

Trying to solve https://github.com/apache/airflow/issues/20088#issue-1072711201

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by potiuk — CHANGES_REQUESTED** [review]
I looked a bit closer and thought about consequences. I really think we shoudl do it in backwards-compatible way and run a a conditional method. We would like to install new google provider with both - older and newer installations where httpx is potentially limited by Airlfow installation (and especially airflow constraints). 

I think the right solution will be to sense with 'inspect signature' https://docs.python.org/3/library/inspect.html#introspecting-callables-with-the-signature-object and use the right parameter. Unfortunately httpx developer did not provide us with backwards-compatibility path, so we have to do it ourselves I am afraid.

**Comment by potiuk** [comment]
ANy chance to get it thsi way @philipherrmann  ?

**Comment by philipherrmann** [comment]
Sure. I can't oversee the consequences nearly as good, so I would just follow your suggestion. Sensing is a nice idea, but just try/except is more "pythonic" (EAFP), isn't it? I would go for that later.

**Comment by potiuk** [comment]
> Sure. I can't oversee the consequences nearly as good, so I would just follow your suggestion. Sensing is a nice idea, but just try/except is more "pythonic" (EAFP), isn't it? I would go for that later.

Both would work, I guess, but I think checking for attr is more explicit IMHO and better expresses the intent. Also same exception can be thrown if you happen to pass another parameter which is unexpected which might hide another error

File: airflow/providers/google/cloud/hooks/cloud_sql.py [hunk 1]
```diff
@@ -769,7 +775,7 @@ def __init__(
 
     @staticmethod
     def _get_bool(val: Any) -> bool:
-        if val == 'False':
+        if val == 'False' or val is False:
             return False
         return True
```
Comment (potiuk) [inline airflow/providers/google/cloud/hooks/cloud_sql.py:780]:
```suggestion
        return val not in ("False", False)
```

**Review by potiuk — APPROVED** [review]
One nit with simpler function.

**Pull request merged**